sw/device/silicon_creator/rom_ext/rom_ext_epmp.c

```diff
@@ -54,3 +54,57 @@ void rom_ext_epmp_unlock_owner_stage_r(epmp_region_t region) {
   CSR_SET_BITS(CSR_REG_PMPCFG0,
                ((kEpmpModeNapot | kEpmpPermLockedReadOnly) << 16));
 }
+
+void rom_ext_epmp_mmio_adjust(void) {
+  const int kEntry = 11;
+  epmp_region_t region = {
+      .start = 0x40000000,
+      .end = 0x50000000,
+  };
+  epmp_state_configure_napot(kEntry, region, kEpmpPermLockedReadWrite);
+
+  // Update the hardware configuration (CSRs).
+  //
+  // Entry is hardcoded as 2. Make sure to modify hardcoded values if changing
+  // kEntry.
+  //
+  // The `pmp11cfg` configuration is the fourth field in `pmpcfg2`.
+  //
+  //            32          24          16           8           0
+  //             +-----------+-----------+-----------+-----------+
+  // `pmpcfg2` = |  `pmp11cfg |  `pmp10cfg`| `pmp9cfg` | `pmp8cfg` |
+  //             +-----------+-----------+-----------+-----------+
+  CSR_WRITE(CSR_REG_PMPADDR11,
+            (uint32_t)region.start >> 2 |
+                ((uint32_t)region.end - (uint32_t)region.start - 1) >> 3);
+  CSR_CLEAR_BITS(CSR_REG_PMPCFG2, 0xff << 24);
+  CSR_SET_BITS(CSR_REG_PMPCFG2,
+               ((kEpmpModeNapot | kEpmpPermLockedReadWrite) << 24));
+}
+
+void rom_ext_epmp_otp_dai_lockout(void) {
```

Comment:
It would be nice to start a doc under `rom_ext/docs/SiVal.md`, with one of the sections capturing the final EPMP configuration. Maybe this can be captured using the print function you added in this PR.